solutions/swb-app/src/datasetRoutes.ts

```diff
@@ -85,6 +87,25 @@ export function setUpDSRoutes(
     })
   );
 
+  // Delete dataset
+  router.delete(
+    '/datasets/:datasetId',
+    wrapAsync(async (req: Request, res: Response) => {
+      const environments: Environment[] = await environmentService.listEnvironmentsForDataSet(
+        req.params.datasetId
+      );
+      if (environments.length) {
+        const existingEnvironments = environments.map((env) => `${env.name}(${env.id})`).join(', ');
+        throw Boom.badRequest(
+          `Cannot delete dataset because it is associated with environments: ${existingEnvironments}`
```

Comment:
I'm thinking this message should be `Cannot delete dataset because it has environments associated with it`. 

We shouldn't return the list of existing environments because this list wil eventually be incorrect. What if there's hundreds of environments associated with a dataset? Will we be able to guarantee that we get all environments?

Reply:
Agree with message change, however I do think we should return at least partial list of environments - otherwise it won't be customer friendly to check every environment whether it has dataset in question

Reply:
Updated, since service signature was changed